operator/pkg/lbipam/lbipam.go

```diff
@@ -0,0 +1,1791 @@
+				if err != nil {
+					return fmt.Errorf("markPoolConflicting: %w", err)
+				}
+				break
+			}
+		}
+	}
+
+	// un-mark pools that no longer conflict
+	for _, poolOuter := range ipam.pools {
+		if !isPoolConflicting(poolOuter) {
+			continue
+		}
+
+		outerRanges, _ := ipam.rangesStore.GetRangesForPool(poolOuter.GetName())
+
+		// If the pool is still internally conflicting, don't un-mark
+		if conflicting, _, _ := areRangesInternallyConflicting(outerRanges); conflicting {
+			continue
+		}
+
+		poolConflict := false
+		for _, poolInner := range ipam.pools {
+			if poolOuter.GetName() == poolInner.GetName() {
+				continue
+			}
+
+			innerRanges, _ := ipam.rangesStore.GetRangesForPool(poolInner.GetName())
+			if conflicting, _, _ := areRangesConflicting(outerRanges, innerRanges); conflicting {
+				poolConflict = true
+				break
+			}
+		}
+
+		// The outer pool, which is marked conflicting no longer conflicts
+		if !poolConflict {
+			err := ipam.unmarkPool(poolOuter)
+			if err != nil {
+				return fmt.Errorf("unmarkPool: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// markPoolConflicting marks the targetPool as "Conflicting" in its status and disables all of its ranges internally.
+func (ipam *LBIPAM) markPoolConflicting(
+	targetPool, collisionPool *cilium_api_v2alpha1.CiliumLoadBalancerIPPool,
+	targetRange, collisionRange *LBRange,
+) error {
+	// If the target pool is already marked conflicting, than there is no need to re-add a condition
+	if isPoolConflicting(targetPool) {
+		return nil
+	}
+
+	ipam.logger.WithFields(logrus.Fields{
+		"pool1-name": targetPool.Name,
+		"pool1-cidr": ipNetStr(targetRange.allocRange.CIDR()),
+		"pool2-name": ipNetStr(collisionRange.allocRange.CIDR()),
+		"pool2-cidr": collisionPool.Name,
+	}).Warnf("Pool '%s' conflicts since CIDR '%s' overlaps CIDR '%s' from IP Pool '%s'",
+		targetPool.Name,
+		ipNetStr(targetRange.allocRange.CIDR()),
+		ipNetStr(collisionRange.allocRange.CIDR()),
+		collisionPool.Name,
+	)
+
+	conflictMessage := fmt.Sprintf(
+		"Pool conflicts since CIDR '%s' overlaps CIDR '%s' from IP Pool '%s'",
+		ipNetStr(targetRange.allocRange.CIDR()),
+		ipNetStr(collisionRange.allocRange.CIDR()),
+		collisionPool.Name,
+	)
+
+	// Mark all ranges of the pool as internally disabled so we will not allocate from them.
+	targetPoolRanges, _ := ipam.rangesStore.GetRangesForPool(targetPool.GetName())
+	for _, poolRange := range targetPoolRanges {
+		poolRange.internallyDisabled = true
+	}
+
+	if ipam.setPoolCondition(targetPool, ciliumPoolConflict, meta_v1.ConditionTrue, "cidr_overlap", conflictMessage) {
+		err := ipam.patchPoolStatus(targetPool)
+		if err != nil {
+			return fmt.Errorf("patchPoolStatus: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// unmarkPool removes the "Conflicting" status from the pool and removes the internally disabled flag from its ranges
+func (ipam *LBIPAM) unmarkPool(targetPool *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) error {
```

Comment:
Would be cool if this became a method on `targetPool` instead.

Reply:
I think we can refactor this at some point, but doing so now would drastically delay the PR yet further, so I would like to postpone that to a separate PR.

Reply:
SGTM